operators/pkg/controller/kibana/pod/pod.go

```diff
@@ -36,12 +36,15 @@ func NewPodTemplateSpec(kb v1alpha1.Kibana) corev1.PodTemplateSpec {
 	objectMeta := kb.Spec.PodTemplate.ObjectMeta.DeepCopy()
 	spec := kb.Spec.PodTemplate.Spec.DeepCopy()
 
-	// add (or override) our labels on top of user-provided ones
 	if objectMeta.Labels == nil {
 		objectMeta.Labels = map[string]string{}
 	}
+	// add our labels on top of user-provided ones,
+	// only if not already specified by the user
 	for k, v := range label.NewLabels(kb.Name) {
-		objectMeta.Labels[k] = v
+		if _, exists := objectMeta.Labels[k]; !exists {
+			objectMeta.Labels[k] = v
+		}
```

Comment:
We have the same bits in the ES controller. I'm fine with that but given it's a bit special (it has 2 lines of comments! ;-)). I wonder if it deserves to live in the `overrides` package to factorize the code and explicit that it's special:

https://github.com/elastic/cloud-on-k8s/blob/85a7eb1ef13f18a7a61dc072648cbb9626b887e4/operators/pkg/controller/elasticsearch/version/common.go#L275-L281

Reply:
Could be: 
```go
objectMeta = overrides.Labels(objectMeta, label.NewPodLabels(es, version, cfg))
```

Reply:
Yeah I agree, will do that.